text/0000-if-let-guard.md

```diff
@@ -0,0 +1,169 @@
+- Feature Name: if_let_guard
+- Start Date: 2018-01-15
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+[summary]: #summary
+
+Allow `if let` guards in `match` expressions.
+
+# Motivation
+[motivation]: #motivation
+
+This feature would greatly simplify some logic where we must match a pattern iff some value computed from the `match`-bound values has a certain form, where said value may be costly or impossible (due to affine semantics) to recompute in the match arm.
```

Comment:
„iff“ -> „if“ typo or is this some sort of abbreviation?

Reply:
*iff* means *if and only if* to distinguish between logical implication `->` or logical equivalence `<->`. Could be written out to make it clearer for those not familiar with this particular mathematical jargon.

Reply:
[If and only if](http://thefreedictionary.com/iff)